### PR TITLE
Improve "index" pages content

### PR DESCRIPTION
**Pull request by melissawm**
# Description
This PR includes a few improvements suggested by the documentation working group:
- Adding context for index pages in each section;
- Removing "duplicated ToCs" from these index pages;
- Removing the "Organization of documentation for napari" document, as it is no longer needed.

cc @neuromusic @tdimino

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

File: docs/guides/index.md [hunk 1]
```diff
@@ -1,6 +1,15 @@
 # In-depth explanations
 
-To learn about using `magicgui` in napari, see the {ref}`magicgui` guide.
+```{note}
+These pages describe advanced usage and how napari works internally. If you are
+just getting started, check out our tutorials or how-to guides instead.
```
Comment (jni) [inline docs/guides/index.md]: @melissawm could you add links for these here?

File: docs/howtos/layers/index.md [hunk 2]
```diff
@@ -7,13 +7,5 @@ following how-to guides:
 ```{tableofcontents}
```
Comment (jni) [inline docs/howtos/layers/index.md:7]: @melissawm do you want to update this one also?

File: docs/tutorials/index.md [hunk 3]
```diff
@@ -1,34 +1,12 @@
...
 These tutorials will help you explore the main usage modes and methods of
 napari, and target people who want to use napari. If you are also interested
-in contributing to napari then please check out [contributing
-guidelines](https://github.com/napari/napari/blob/master/docs/developers/contributing.md).
-
-For help installing napari checkout our
-[installing napari](./fundamentals/installation) tutorial. If this is your first
-contact with napari, then begin with our
-[getting started](./fundamentals/getting_started) tutorial.
+in contributing to napari then please check out [Community](../community/index).
```
Comment (jni) [inline docs/tutorials/index.md]:
```suggestion
in contributing to napari then please check out [the Community page](../community/index).
```

File: docs/tutorials/index.md [hunk 4]
```diff
@@ -1,34 +1,12 @@
...
-[napari/napari.github.io](https://github.com/napari/napari.github.io). If as
-you're going through the tutorials you spot any errors or can think of ways to
-improve them please raise an issue on the repository or make a PR, we'd love to
-have the community help make them better for everyone.
-
-[napari tutorials home page](http://www.napari.org/tutorials)
+Our tutorials are hosted on [the napari Github repository](https://github.com/napari/napari).
+If as you're going through the tutorials you spot any errors or can think of
+ways to improve them please raise an issue on the repository or make a pull
+request, we'd love to have the community help make them better for everyone.
```
Comment (jni) [inline docs/tutorials/index.md]:
```suggestion
As you make your way through the tutorials, if you spot any errors or areas that
can be improved, please raise an issue on the repository or make a pull
request! We love it when our community helps make them better for everyone!
To see how to modify documentation pages, please see our
[documentation contributing guide](../developers/documentation/index).
```

**Review by jni — APPROVED** [review]
Thank you @melissawm! I found one case where you still left a `{tableofcontents}` directive, and made some minor suggestions, but otherwise I'm happy with these changes!

File: docs/howtos/layers/index.md (hunk 2 above)
Comment (melissawm) [reply docs/howtos/layers/index.md:7]: I was actually going to keep this one as I think it's useful in this page. Do you have another suggestion for linking to these sub-pages?

File: docs/guides/index.md (hunk 1 above)
Comment (melissawm) [reply docs/guides/index.md]: Done!

**Review by sofroniewn — APPROVED** [review]
Looks great, thanks very much!!

**Review by Carreau — APPROVED** [review]

**Pull request merged**